### ISSUE TITLE
Generalize StructureAlignment Visualization

### ISSUE THREAD

**Issue by lafita** [biojava/biojava#396]
Related to #393. The idea is to extract _selectors_ from the alignment data structures in BioJava (**AFPChain** and **MultipleAlignment**) so that any molecular viewer can represent structure alignments. A good solution could be to extract **ResidueNumbers**, which can then be serialized and parsed by the viewers.

**Comment by lafita** [comment]
Steps to be made:
1) All classes that are **Jmol** specific should be moved to **structure-gui** module and must contain the word **Jmol** in their name.
2) All methods and classes that are not **Jmol** specific should be moved to the **structure** module, should never contain the word **Jmol** in their name and should contain the word **Display** (to differentiate from **Utils** and **Tools**).
3) The **structure-gui** module should contain as little algorithmic part as possible (only build the BioJava-Jmol connection), and all the current features implemented should be able to be reproduced by calling methods in the **Display** classes of the **structure** module.

**Comment by sbliven** [comment]
StructureIdentifier already provides more-or-less the residue-based *selector* functionality you're describing. You would just need to write a method that converts a SubstructureIdentifier into a Jmol selection string.

This also nicely complements the long-range plans discussed in #81 for a successor to SubstructureIdentifier that would be capable of arbitrary selections, including additional properties and basic set operations.

**Comment by lafita** [comment]
Yes, it is currently what we are using for symmetric repeats in internal symmetry. 
For alignments it would be a bit more involved to include gaps in the **Identifier Ranges**, but I like the idea of a method to convert a **StructureIdentifier** into a **Jmol** selector (or any kind of selector).

**Comment by andreasprlic** [comment]
+1
